src/ray/core_worker/reference_count.h

```diff
@@ -336,7 +382,7 @@ class ReferenceCounter {
   /// inlined dependencies are inlined or when the task finishes for plasma
   /// dependencies.
   void RemoveSubmittedTaskReferences(const std::vector<ObjectID> &argument_ids,
-                                     std::vector<ObjectID> *deleted)
+                                     bool release_lineage, std::vector<ObjectID> *deleted)
```

Comment:
Consider always accounting for lineage, to remove this flag.

Reply:
I changed it to always account for lineage, but we still need this flag because the TaskManager needs to communicate which tasks are retryable or not.